w1r3/java/src/main/java/W1R3.java

```diff
@@ -0,0 +1,404 @@
+    private int workers;
+
+    @Option(names = "-transport", description = "the transports (JSON, GRPC+CFE, GRPC+DP) used by the benchmark")
+    private String[] transports;
+
+    @Option(names = "-object-size", description = "the object sizes used by the benchmark")
+    private Integer[] objectSizes;
+
+    public static void main(String... args) {
+        var exitCode = new CommandLine(new W1R3()).execute(args);
+        System.exit(exitCode);
+    }
+
+    @Override
+    public Integer call() throws Exception {
+        if (transports == null || transports.length == 0) {
+            transports = new String[] { "JSON", "GRPC+CFE", "GRPC+DP" };
+        }
+        if (objectSizes == null || objectSizes.length == 0) {
+            objectSizes = new Integer[] { 100 * KB, 2 * MiB, 100 * MB };
+        }
+        var instance = UUID.randomUUID().toString();
+
+        System.out.printf("## Starting continuous GCS Java SDK benchmark\n");
+        System.out.printf("# object-sizes: %s\n", Arrays.toString(this.objectSizes));
+        System.out.printf("# transports: %s\n", Arrays.toString(this.transports));
+        System.out.printf("# project-id: %s\n", this.project);
+        System.out.printf("# bucket: %s\n", this.bucket);
+        System.out.printf("# deployment: %s\n", this.deployment);
+        System.out.printf("# instance: %s\n", instance);
+        System.out.printf("# Java SDK Version %s\n", sdkVersion);
+        System.out.printf("# HTTP Version %s\n", httpClientVersion);
+        System.out.printf("# Protobuf Version %s\n", protobufVersion);
+        System.out.printf("# gRPC Version %s\n", grpcVersion);
+
+        var random = new SecureRandom();
+        var randomData = makeRandomData(this.objectSizes, random);
+
+        var otelSdk = setupOpenTelemetrySdk(instance);
+
+        var clients = makeClients(transports);
+        var uploaders = makeUploaders();
+        var workers = new ArrayList<Thread>();
+        for (int i = 0; i != this.workers; ++i) {
+            workers.add(new Thread(
+                    (() -> {
+                        worker(clients, uploaders, randomData, random.nextInt(), instance, otelSdk);
+                    })));
+        }
+        for (var t : workers)
+            t.start();
+        for (var t : workers)
+            t.join();
+        return 0;
+    }
+
+    public static class Transport {
+        public Transport(String n, Storage c) {
+            name = n;
+            client = c;
+        }
+
+        public final String name;
+        public final Storage client;
+    }
+
+    public static interface Uploader {
+        public BlobId upload(Storage client, BlobInfo info, ByteBuffer input) throws Exception;
+
+        public String name();
+    }
+
+    private void worker(
+            Transport[] transports, Uploader[] uploaders,
+            byte[] randomData, int seed, String instance,
+            OpenTelemetrySdk otelSdk) {
+        var random = new Random(seed);
```

Comment:
Not sure if important in your opinion, but the distribution of Random is not uniform. If we want it to be uniform we'll need to add another dependency and update the code in a later PR.

Reply:
Yeah, we should fix that in a separate pass. #64